### PR TITLE
Use Azurite to test Azure Storage

### LINKED ISSUE #820: Use Azurite to test Azure Storage
**Is your feature request related to a problem? Please describe.**
At the moment we skip the Azure Storage tests in our travis builds.

**Describe the solution you'd like**
Idea stolen from @chriscoffee 
We could use Azurite https://github.com/Azure/Azurite to test the storage just like we test s3 using minio.

**Describe alternatives you've considered**
Using Azure Blob Storage.

### PR DESCRIPTION
**Pull request by chriscoffee**
**What is the problem I am trying to address?**
At the moment we skip the Azure Storage tests in our travis builds. We could use Azurite https://github.com/Azure/Azurite to test the storage just like we test s3 using minio.

**How is the fix applied?**

Whilst we wait for the interface to be implemented so we can enable the tests #703. I've updated the compose file to include azurite and also sorted the docker-compose file so it doesn't have a mixture of  array/dicts for environment variable defintions and as a list for command. Also sorted alphabetically.

- [x] Add Azurite to docker-compose
- [ ] Add tests against local Azure Blob Storage

Signed-off-by: Chris M <millscj01@gmail.com>

Fixes #820

### CODE REVIEW

**Comment by arschles** [comment]
Doh ignore my question, I just re-read your checklist in the OP

**Comment by arschles** [comment]
@chriscoffee let me know if you want some help on this. I'm happy to collaborate on writing some of th tests

**Comment by chriscoffee** [comment]
@arschles cheers for the offer, already mocked out what I think they will look like once #703 is done. I'll check where it's at. 👍

File: docker-compose.yml [hunk 1]
```diff
@@ -1,40 +1,47 @@
 version: '3'
 services:
+  azurite:
```
Comment (arschles) [inline docker-compose.yml]: @chriscoffee should we configure the unit tests to talk to azurite?

Also https://github.com/gomods/athens/pull/973 could use this as well 😄

**Comment by arschles** [comment]
@chriscoffee this is super close. still want to take this? if not, I'm happy to pick it up. I'll ping you in slack too

**Comment by chriscoffee** [comment]
> @chriscoffee this is super close. still want to take this? if not, I'm happy to pick it up. I'll ping you in slack too

I'll have a look at this later today @arschles  👍 

Kinda forgot about it... My bad mate

**Comment by arschles** [comment]
not a problem in the world @chriscoffee - thanks for answering! this will help @Zyqsempai out a lot on https://github.com/gomods/athens/issues/995

**Comment by arschles** [comment]
@chriscoffee we could point the azure blob tests in a follow up? let me know what you think - not a problem in the world either way, but I do want to get @Zyqsempai going on https://github.com/gomods/athens/issues/995

**Comment by arschles** [comment]
@chriscoffee said it's cool to merge this and we will add the tests in a follow-up PR. thanks Chris!

**Pull request merged**